subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -12,11 +12,46 @@
 
 LOG_MODULE_REGISTER(dfu_target_mcuboot, CONFIG_DFU_TARGET_LOG_LEVEL);
 
+#define MAX_FILE_SEARCH_LEN 500
 #define MCUBOOT_HEADER_MAGIC 0x96f3b83d
 
 static struct flash_img_context flash_img;
 static size_t offset;
 
+int dfu_ctx_mcuboot_set_b1_file(char *file, bool s0_active, char **update)
+{
+	if (file == NULL   ||  update == NULL) {
+		return -EINVAL;
+	}
+
+	/* Ensure that 'file' is null-terminated. */
+	if (strnlen(file, MAX_FILE_SEARCH_LEN) == MAX_FILE_SEARCH_LEN) {
+		return -EINVAL;
+	}
+
+	/* We have verified that there is a null-terminator, so this is safe */
+	char *space = strstr(file, " ");
+
+	if (space == NULL) {
+		/* Could not find space separator in input */
+		*update = NULL;
+
+		return 0;
+	}
+
+	if (s0_active) {
+		/* Point after space to 'activate' second file path (S1) */
+		*update = space + 1;
+	} else {
+		*update = file;
+
+		/* Insert null-terminator to 'activate' first file path (S0) */
+		*space = '\0';
+	}
```

Comment:
You could consider `strtok()`

    char *strtok_state;
    char *path1 = strtok_r(file, " ", &strtok_state); // First path (sets space to \0)
    char *path2 = strtok_r(file, " ", &strtok_state); // Second path (NULL if no space was found)
    *update = (!path2 || s0_active) ? path2 : path1; // Return correct path or NULL if path2 is NULL.
    return 0;

Reply:
AFAICS there is no memory safe variant of `strtok` where I can provide the max number of characters it should search for the delimeter

Reply:
Yes, you would need to keep the part where you check for the existence of the `'\0'`. This was meant as a replacement for what comes after that. It's up to you :)

Reply:
Ah, my bad. Will see if it fits with the API.

Reply:
`strstr` is already in use in `download_client`, while `strtok_r` would have to be added (increasing code size). Also i find the `strstr` variant easier to read. (Also I don't find it in our  libc?)